File: QuizGame-WebAPI/Models/Game.cs
namespace QuizGame.Models;

public class Game
{
    public int Id {get; set;}
    public string Name {get; set;}
    public int PassingScore {get; set;}
    public DateTime DueDate {get; set;}
    public Quiz Quiz {get; set;}
    public IEnumerable<QuizGameUser>? AssignedUsers {get; set;}
    public QuizGameUser? Owner {get; set;}
    
    internal Game()
    {
        
    }
    public Game(string name, Quiz quiz)
    {
        Name = name;
        Quiz = quiz;
    }
}

File: QuizGame-Client/src/app/models/game.ts
import { FormControl } from "@angular/forms";
import { User } from "./user";

export interface Game{
    id: number;
    name: string;
    passingScore: number;
    dueDate: Date;
    quizId: number;
    quizName: string;
    assignedUsers: User[];
    owner: User;
}

export interface GameForm{
    id?: FormControl<number|null>;
    name: FormControl<string|null>;
    description: FormControl<string|null>;
}